### PR TITLE
Improve gv output

### PR DESCRIPTION
**Pull request by snim2**
**Needs squashing**

This PR improves the mapping and edit script visualisations that are provided in Graphviz format (for debugging), in three ways:

  * Borders are drawn between the source and destination ASTs
  * The two borders are lined up
  *  Whitespace nodes now have their whitespace made visible (e.g. "\n" now appears in the graphs).

### Example

On master branch:

  * [map_master.pdf](https://github.com/softdevteam/diffract/files/1382661/map_master.pdf)
  * [edit_master.pdf](https://github.com/softdevteam/diffract/files/1382662/edit_master.pdf)

On this branch:

  * [map.pdf](https://github.com/softdevteam/diffract/files/1382664/map.pdf)
  * [edit.pdf](https://github.com/softdevteam/diffract/files/1382665/edit.pdf)

### CODE REVIEW

File: src/lib/emitters.rs [hunk 1]
```diff
@@ -378,6 +392,7 @@ impl RenderDotfile for MappingStore<String> {
                            common);
             digraph.push(line);
         }
+        digraph.push(String::from("\t{ rank=same; FROM0; TO0; }\n"));
```
Comment (ltratt) [inline src/lib/emitters.rs]: Is this intentionally hard-coded between two nodes?

Comment (snim2) [reply src/lib/emitters.rs]: For now, yes. At the moment we assume that the root node is always node zero. This isn't entirely a safe assumption, as the edit script generator can replace the root node. This needs some careful thought though (and a new PR) so I have raised #41

Comment (ltratt) [reply src/lib/emitters.rs]: OK.

**Comment by ltratt** [comment]
OK, I like this. Shall we squash?

**Comment by snim2** [comment]
Yes please! Squash coming up...

**Comment by snim2** [comment]
Squashed.

**Comment by snim2** [comment]
About to bring this up to date with master...

**Comment by snim2** [comment]
Done

**Pull request merged**